Correctly use makeNonNull for receivers and arguments in method invocations (#306)
eisop/checker-framework: docs/manual/nullness-checker.tex

  checker will unsoundly assume that the argument to \<Map.get> is a key
  for the receiver map.  It will not do any checking of
  \refqualclass{checker/nullness/qual}{KeyFor} and related qualifiers.

\item
  \label{nullness-conservativeArgumentNullnessAfterInvocation}%

>   If you supply the \code{-AconservativeArgumentNullnessAfterInvocation} command-line option, then the

Reviewer: Update this text similarly to the changelog changes I suggested.

Author: fixed

  checker will assume receiver and params are nullable after the method invocation. When running gradle

>   tests we disable this option by default except when doing \code{checkNullness} test.

Reviewer: Is this important to readers of the manual?

Author: No, as the task `checkNullness` is for our own testing purpose. I will delete this line.

  checker will unsoundly assume that the argument to \<Map.get> is a key
  for the receiver map.  It will not do any checking of
  \refqualclass{checker/nullness/qual}{KeyFor} and related qualifiers.

\item
  \label{nullness-conservativeArgumentNullnessAfterInvocation}%

>   Applying the `-AconservativeArgumentNullnessAfterInvocation` improves the soundness of

Reviewer: ```suggestion
  If you supply the `-AconservativeArgumentNullnessAfterInvocation` command-line option, the soundness of
```

>   the Nullness Checker. In previous versions and without specifying the new flag, the receiver and

Reviewer: ```suggestion
  the Nullness Checker is improved. In previous versions and without supplying the new flag, the receiver and
```

  arguments that are passed to non-null parameters in a method call
  or constructor invocation are assumed to be non-null after the invocation.
  This assumption is unsound in general, but holds for most code.

>   Use the new flag to soundly handle the nullness of the receiver and arguments in an invocation.

Reviewer: ```suggestion
  Use this flag to soundly handle the nullness of the receiver and arguments in an invocation.
```
